table-api/src/main/java/io/deephaven/api/ColumnMatch.java

```diff
@@ -0,0 +1,33 @@
+package io.deephaven.api;
+
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Parameter;
+
+@Immutable(builder = false, copy = false)
+public abstract class ColumnMatch implements JoinMatch {
+
+    public static ColumnMatch of(ColumnName left, ColumnName right) {
+        return ImmutableColumnMatch.of(left, right);
+    }
+
+    public static ColumnMatch parse(String x) {
```

Comment:
We probably need to `trim` our column names. See `com.illumon.iris.db.tables.select.MatchPairFactory`.

Reply:
I've separated out calls to `of` vs `parse`.